src/bin/CMakeLists.txt

```diff
@@ -7,11 +7,199 @@ else()
     set(SOURCES linux/init.c)
 endif()
 
-add_library(msquic SHARED ${SOURCES})
+if(BUILD_SHARED_LIBS)
+    add_library(msquic SHARED ${SOURCES})
+    target_link_libraries(msquic PRIVATE core platform inc warnings)
+else()
+    add_library(msquic_static STATIC static/init.c)
+    target_link_libraries(msquic_static PRIVATE core platform inc warnings)
+    target_compile_definitions(msquic_static PUBLIC QUIC_BUILD_STATIC)
+    set_property(TARGET msquic_static PROPERTY FOLDER "${QUIC_FOLDER_PREFIX}libraries")
+
+    # We need to take given library and walk all its dependencies, including
+    # transient dependencies. For each dependency, if it either compiles
+    # directly to or imports a static library, that library needs to be
+    # added to the QUIC_STATIC_LIBS list. This list is ultimately fed into
+    # the archiving utility to produce a monolithic static library as
+    # static library flags
+    set(EXCLUDE_LIST "ws2_32;schannel;ntdll;bcrypt;ncrypt;crypt32;iphlpapi;advapi32")
```

Comment:
Is there a way to get a list of all the libraries that get included, but aren't in the exclude list already?

Reply:
I guess I can compile it locally and uncomment the line below...

Reply:
Yea if you want we can emit that log always. There are a few lines I left in there to make it easier to debug.

Reply:
One comment I can make about the implementation of this function btw is that the reason we use `GLOBAL PROPERTY` registers to track the data is because of how lexical scope rules work in CMake w.r.t. function recursion.

Reply:
Since these libraries are listed 3 times (once in inc, and twice in this file) I'd like to see if there is any way we can unify them. That we if we ever have to update 1, we update all 3 at the same time without missing any.

Reply:
The change I was about to make was to simply exclude `inc` which I believe might work?

Reply:
In theory yes that should.

Reply:
OK that worked. Also you can see the full final link command by looking at the contents of the file `src\bin\Debug\quicdeps.txt` for debug builds for example from the build root.

Reply:
I resolved this by removing inc (and warnings) from the intermediate target that globbed the libraries. Since there's no code in this target, we can remove warnings. The inc target meanwhile is included in the final `msquic` target. This meant I could remove the entire `EXCLUDE_LIST` variable altogether.